x/metadata/keeper/scope.go

```diff
@@ -281,20 +247,55 @@ func (k Keeper) ValidateScopeUpdate(ctx sdk.Context, existing, proposed types.Sc
 
 // ValidateScopeRemove checks the current scope and the proposed removal scope to determine if the the proposed remove is valid
 // based on the existing state
-func (k Keeper) ValidateScopeRemove(ctx sdk.Context, existing, proposed types.Scope, signers []string) error {
-	// IDs must match
-	if len(existing.ScopeId) > 0 {
-		if !proposed.ScopeId.Equals(existing.ScopeId) {
-			return fmt.Errorf("cannot update scope identifier. expected %s, got %s", existing.ScopeId, proposed.ScopeId)
-		}
+func (k Keeper) ValidateScopeRemove(ctx sdk.Context, scope types.Scope, signers []string) error {
+	if err := k.ValidateAllPartiesAreSigners(scope.Owners, signers); err != nil {
+		return err
 	}
-	if err := k.ValidateAllPartiesAreSigners(existing.Owners, signers); err != nil {
+
+	if err := k.validateScopeUpdateValueOwner(ctx, scope.ValueOwnerAddress, "", signers); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (k Keeper) validateScopeUpdateValueOwner(ctx sdk.Context, existing, proposed string, signers []string) error {
+	// If they're the same, we don't need to do anything.
+	if existing == proposed {
+		return nil
+	}
+	if len(existing) > 0 {
+		if k.AccountIsMarker(ctx, existing) {
```

Comment:
since `k.AccountIsMarker(ctx, existing)` does read the state store, and is used late(potentially), maybe store it?

Reply:
oh sorry did not see this was a draft :(

Reply:
All good! It's a good call and kind of nicer to change before putting it into ready for review!

Reply:
I just double checked. It's actually checking two different values. `AccountIsMarker` is called once for `existing` and once for `proposed`. There's a short-circuit at the top when the two are equal, so we know it's not making the same check twice. The calls to `HasSignerWithMarkerValueAuthority` are similar with the added difference of needed different permissions.

Reply:
Makes sense, thanks for verifying!

Reply:
You were right about being able to save a lookup. Both `k.AccountIsMarker(ctx, existing)` and `k. HasSignerWithMarkerValueAuthority(ctx, existing, ...)` were doing the same marker account lookup. So I combined them into one function so they can both use the same thing.